spec/checkMergeConflictSpec.js

```diff
@@ -0,0 +1,216 @@
+      },
+    };
+
+    robot = createProbot({
+      id: 1,
+      cert: 'test',
+      githubToken: 'test',
+    });
+
+    app = robot.load(oppiaBot);
+    spyOn(app, 'auth').and.resolveTo(github);
+    spyOn(
+      checkMergeConflictModule,
+      'checkMergeConflictsInAllPullRequests'
+    ).and.callThrough();
+    spyOn(
+      checkMergeConflictModule,
+      'checkMergeConflictsInPullRequest'
+    ).and.callThrough();
+  });
+
+  describe('when pull request gets merged', () => {
+    beforeEach(async () => {
+      // Simulate a merged conflict in pull request.
+      payload.payload.pull_request.merged = true;
+      payload.payload.action = 'closed';
+
+      const pullRequest = { ...payload.payload.pull_request };
+      pullRequest.merged = false;
+      pullRequest.mergeable = false;
+
+      github.pulls = {
+        get: jasmine.createSpy('get').and.resolveTo({
+          data: pullRequest,
+        }),
+        list: jasmine.createSpy('list').and.resolveTo({
+          data: [pullRequest],
+        }),
+      };
+
+      await robot.receive(payload);
+    });
+
+    it('should check for merge conflict', () => {
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInAllPullRequests
+      ).toHaveBeenCalled();
+      expect(
+        checkMergeConflictModule.checkMergeConflictsInPullRequest
+      ).toHaveBeenCalled();
+    });
+
+    it('ping pr author regarding merge conflict', () => {
+      expect(github.issues.createComment).toHaveBeenCalled();
+
+      const link = 'link'.link(
+        'https://help.github.com/articles/resolving-a-merge-conflict-using-the-command-line/'
+      );
+      const params = {
+        repo: payload.payload.repository.name,
+        owner: payload.payload.repository.owner.login,
+        number: payload.payload.pull_request.number,
+        body:
+          'Hi @' +
+          payload.payload.pull_request.user.login +
+          '. Due to recent changes in the "develop" branch, ' +
+          'this PR now has a merge conflict. ' +
+          'Please follow this ' +
+          link +
+          ' if you need help resolving the conflict, ' +
+          'so that the PR can be merged. Thanks!',
+      };
+      expect(github.issues.createComment).toHaveBeenCalledWith(params);
+    });
+
+    it('adds merge conflict label', () => {
+      expect(github.issues.addLabels).toHaveBeenCalled();
+      const params = {
+        repo: payload.payload.repository.name,
+        owner: payload.payload.repository.owner.login,
+        number: payload.payload.pull_request.number,
+        labels: [mergeConflictLabel.name],
+      };
+      expect(github.issues.addLabels).toHaveBeenCalledWith(params);
+    });
+  });
+
+  describe("when pull request that doesn't cause merge conflict gets merged", () => {
+    beforeEach(async () => {
+      // Simulate a merged conflict in pull request.
+      payload.payload.pull_request.merged = true;
+      payload.payload.action = 'closed';
```

Comment:
Add a comment to explain the purpose of this.

Reply:
The same comment is for both lines (this and the one above).